### PR TITLE
doxygen: Fixed all warnings

### PR DESCRIPTION
**Pull request by tsparber**
Congratulations you fixed 134 of doxygen's warnings (old: 134 new: 0).

This pull request fixes all open doxygen warnings, by fixing the code comments and documentation.

**Documentation:**
This PR tries to also fix the documentations of sicslowpan and uip6 to reflect the current code state. I’m not sure if everything makes sense.

sicslowpan-doc.txt: There are still some references to HC01, can this be replaced by HC06?
uip6-doc.txt: Section about timers has changed, is this correct?

Doxyfile: The documentations reference static functions, to link to them correctly EXTRACT_STATIC = YES is needed.

**Code:**
I tried to reduce the whitespace changes. All trailing spaces and mixed tab/space conversions (hand corrected) for the affected files can be found in a separate branch ([fix-doxygen-whitespaces](https://github.com/tsparber/contiki/tree/fix-doxygen-whitespace)).
Should I create a separate pull request or merge it into this?

### CODE REVIEW

File: cpu/avr/dev/lanc111.c [hunk 1]
```diff
@@ -1121,9 +1118,8 @@ static int NicPutPacket(NETBUF * nb)
 }
 #endif
 
-/*! \fn NicRxLanc(void *arg)
+/*!
  * \brief NIC receiver thread.
- *
  */
```
Comment (cmorty) [inline cpu/avr/dev/lanc111.c]: This should also go behind the if `#if 1`

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h [hunk 2]
```diff
@@ -20,10 +20,13 @@
  * that point to the correct location in the Info Blocks.  (This is the
  * extern, the actual definition is found in hal/micro/cortexm3/token.c)
  *
- * \param name: The name of the token.
- *
- * \param TOKEN_##name##_ADDRESS: The address in EEPROM at which the token
- * will be stored.  This parameter is generated with a macro above.
```
Comment (cmorty) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: This documentation should be fixed rather then removed. There are too many people who don't know the CPP well enaugh. Maybe add a todo.

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h [hunk 3]
```diff
@@ -33,32 +36,28 @@
 /**
  * \brief Macro for translating token definitions into size variables.
  * This provides a convenience for abstracting the 'sizeof(type)' anywhere.
- *
- * \param name: The name of the token.
- *
- * \param type: The token type.  The types are found in token-stack.h.
  */
```
Comment (cmorty) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: Maybe it's better to just remove TOKEN_MFG from Doxygen, but keep the documentation. Without understanding the code the documentation is not helpful. And explaining what's happening is IMO not worth the trouble.

File: platform/avr-ravenusb/cdc_task.c [hunk 4]
```diff
@@ -75,6 +75,10 @@
 #include <avr/wdt.h>
 #include <util/delay.h>
 
+#if UIP_CONF_IPV6_RPL
+#include "rpl.h"
+#endif
+
```
Comment (cmorty) [inline platform/avr-ravenusb/cdc_task.c]: Why is this needed up here? Maybe add a comment, otherwise someone might move it down again.

File: platform/avr-ravenusb/rng.c [hunk 5]
```diff
@@ -7,6 +7,11 @@
 #include <util/delay.h>
 #include "contiki.h"
 
+#ifdef DOXYGEN_SHOULD_SKIP_THIS
+#define RNG_CONF_USE_ADC            1
+#define RNG_CONF_USE_RADIO_CLOCK    1
+#endif
```
Comment (cmorty) [inline platform/avr-ravenusb/rng.c]: Why do you need to define new macros?

File: platform/wismote/dev/acc-sensor.c [hunk 6]
```diff
@@ -42,8 +42,8 @@ static void
 activate(void)
 {
   /* This assumes that some other sensor system already did setup the ADC
-  /* (in the case of the sky platform it is sensors_light_init that does it)
-
+   * (in the case of the sky platform it is sensors_light_init that does it) */
+  #if 0
```
Comment (cmorty) [inline platform/wismote/dev/acc-sensor.c]: This does not seem documentation related.

File: platform/wismote/dev/acc-sensor.c [hunk 7]
```diff
@@ -52,37 +52,40 @@ activate(void)
...
+  acc_value = 0;
+  #endif
   active = 0;
 }
 /*---------------------------------------------------------------------------*/
 static int
 value(int type)
 {
-/*
+  #if 0
```
Comment (cmorty) [inline platform/wismote/dev/acc-sensor.c]: Same here. It changes the code.

File: platform/wismote/dev/ext-sensor.c [hunk 8]
```diff
@@ -43,8 +43,10 @@ static uint8_t active;
 static int
 value(int type)
 {
-  /* ADC0 corresponds to the port under the logo, ADC1 to the port over the logo,
-     ADC2 and ADC3 corresponds to port on the JCreate bottom expansion port)
+  #if 0
+  /* ADC0 corresponds to the port under the logo,
```
Comment (cmorty) [inline platform/wismote/dev/ext-sensor.c]: same as acc-sensor: Why disable the code?

**Comment by cmorty** [comment]
Awesome peace of work.

File: cpu/avr/dev/lanc111.c (hunk 1 above)
Comment (tsparber) [inline cpu/avr/dev/lanc111.c]: fixed with amended commit

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h (hunk 2 above)
Comment (tsparber) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: The #define TOKEN_MFG is used 3 times, I tried to fix the documentation in the first occurrence and did not copy it to the other two instances.

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h (hunk 3 above)
Comment (tsparber) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: I copied the docs from the first TOKEN_MFG instance now. Should I still remove TOKEN_MFG from doxygen?

File: platform/avr-ravenusb/cdc_task.c (hunk 4 above)
Comment (tsparber) [inline platform/avr-ravenusb/cdc_task.c]: I added the following comment:
```
// Include needs to be up here instead of embedded in the other
// UIP_CONF_IPV6_RPL block, as doxygen seems not to be compatible with
// #includes embedded in other functions and spits out a warning.
```
The warning:
`contiki/platform/avr-ravenusb/cdc_task.c:582: warning: include file rpl.h not found, perhaps you forgot to add its directory to INCLUDE_PATH?`
As moving the #include to another location fixes the warning, it seems not to be a INCLUDE_PATH problem.

File: platform/avr-ravenusb/rng.c (hunk 5 above)
Comment (tsparber) [inline platform/avr-ravenusb/rng.c]: As far as my test goes doxygen seems to have problems with multiple defined statements (`defined(ADMUX) && defined(ADCSRA) && defined(ADCSRB) && ...` )

`contiki/platform/avr-ravenusb/rng.c:25: warning: preprocessing issue while doing constant expression evaluation: syntax error`

Looking at the code again, I noticed that both macros depend on each other. So only either `RNG_CONF_USE_ADC` or `RNG_CONF_USE_RADIO_CLOCK` should be defined.
I opted for `RNG_CONF_USE_RADIO_CLOCK` now as this is also defined in [platform/avr-ravenusb/contiki-conf.h#L94](https://github.com/contiki-os/contiki/blob/master/platform/avr-ravenusb/contiki-conf.h#L94)

File: platform/wismote/dev/acc-sensor.c (hunk 6 above)
Comment (tsparber) [inline platform/wismote/dev/acc-sensor.c]: Doxygen has problems with the not closed nested block comments. As far as I understand the current code, this block is also not active at the moment [platform/wismote/dev/acc-sensor.c#L42](https://github.com/contiki-os/contiki/blob/master/platform/wismote/dev/acc-sensor.c#L42) (Nested comments have the closing tag removed)
To fix the doxygen warnings I closed all comments with the respective end tag and added a `#if 0` to get the same behavior as before, as nested `/* */` comments don't work either (http://c-faq.com/misc/nestcomment.html).

File: platform/wismote/dev/acc-sensor.c (hunk 7 above)
Comment (tsparber) [inline platform/wismote/dev/acc-sensor.c]: I changed the "disable code" style from `/* */` to `#if 0 #endif` to match the rest of the file.

File: platform/wismote/dev/ext-sensor.c (hunk 8 above)
Comment (tsparber) [inline platform/wismote/dev/ext-sensor.c]: The code is also not active at the moment ([platform/wismote/dev/ext-sensor.c#L44](https://github.com/contiki-os/contiki/blob/master/platform/wismote/dev/ext-sensor.c#L44)) and I changed the "disable code" style to match the rest of the file.

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h (hunk 3 above)
Comment (cmorty) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: I don't know. I think it's not worth the trouble.

File: platform/avr-ravenusb/cdc_task.c (hunk 4 above)
Comment (cmorty) [inline platform/avr-ravenusb/cdc_task.c]: That's perfect!

File: platform/wismote/dev/acc-sensor.c (hunk 6 above)
Comment (cmorty) [inline platform/wismote/dev/acc-sensor.c]: Oh, I'm sorry. I just scanned over the changes. Of course the `#if 0` is much better than the block-comment!

File: platform/wismote/dev/acc-sensor.c (hunk 7 above)
Comment (cmorty) [inline platform/wismote/dev/acc-sensor.c]: Dto. Don't know what I was (not) thinking!

**Comment by cmorty** [comment]
@simonduq, @g-oikonomou  As this is related to #1293: Can I get you interested in this PR?

**Comment by simonduq** [comment]
Nice work, thanks!
First comment from a first glance at the diff: I don't like the many new `DOXYGEN_SHOULD_SKIP_THIS`. Creates confusion IMO (which defeats the purpose of Doxygen!), especially when used in combination with other config macros, such as in `#if !UIP_CONF_ROUTER || DOXYGEN_SHOULD_SKIP_THIS`.
Something else: we probably also want to simplify the travis script to stop counting warnings but rather tolerate warning-free builds only.

**Comment by tsparber** [comment]
Thanks for the comments, I reduced the new `DOXYGEN_SHOULD_SKIP_THIS` defines and cleaned up the travis script.

File: cpu/stm32w108/hal/micro/cortexm3/mfg-token.h (hunk 3 above)
Comment (tsparber) [inline cpu/stm32w108/hal/micro/cortexm3/mfg-token.h]: `TOKEN_MFG` is now excluded from doxygen.

**Comment by simonduq** [comment]
Great, thanks a lot for this contribution! :+1: for me
I have a couple very minor comments:
* `You increased the number of warnings caused by doxygen` sounds strange, now that we are warning-free.
* For all the added `#if 0` I would add `#endif /* 0 */` instead of merely `#endif`. Same with  `#if 1`.

**Comment by tsparber** [comment]
I changed the wording to `Doxygen caused some warnings. Please fix these.` ok? or any suggestion?
(and also added the `#endif /* 0 */`)

Is there any interest in fixing all the tabs/white spaces in the files affected by this pull request? (https://github.com/contiki-os/contiki/compare/master...tsparber:fix-doxygen-whitespace) I would then create a separate pull request.

**Comment by simonduq** [comment]
Great thanks. A separate PR would be perfect. Scheduling this for merge in a month (but others are very welcome to merge this earlier)

**Comment by alignan** [comment]
:+1:
